### PR TITLE
per-tier voting accounting.

### PR DESCRIPTION
**Pull request by mejango**
in addition to the standard `_getVotingUnits` exposed by the `Votes` contract for easy Governor use, this PR adds a `_getTierVotingUnits` and tracks snapshotted values accordingly to Governor contracts to be written that use per-tier voting units. Meaning an on-chain governance contract can be written that solicits votes only from holders of a particular tier.

### CODE REVIEW

File: contracts/JBTiered721Delegate.sol [hunk 1]
```diff
@@ -26,13 +26,18 @@ import './libraries/JBIpfsDecoder.sol';
...
   //*********************************************************************//
   // --------------------------- custom errors ------------------------- //
   //*********************************************************************//
 
   error NOT_AVAILABLE();
   error OVERSPENDING();
 
+  mapping(address => mapping(uint256 => address)) private _tierDelegation;
+  mapping(address => mapping(uint256 => Checkpoints.History)) private _delegateTierCheckpoints;
+  mapping(uint256 => Checkpoints.History) private _totalTierCheckpoints;
```
Comment (xBA5ED) [inline contracts/JBTiered721Delegate.sol:39]: Should we make the above mappings public so we can extend the functionality from external contracts (ex. tier governance)? I don't see the data being exposed anywhere

Comment (mejango) [reply contracts/JBTiered721Delegate.sol:39]: Yeah. I just took the pattern from the base class. Probably missed the view fns

**Comment by mejango** [comment]
this looks good to me. 
makes for extra gas on transfers when using voting units, but enables this kinda dank game i wanna build.

**Comment by mejango** [comment]
@drgorillamd review?

**Comment by mejango** [comment]
gonna merge in to get the gist into the latest deploy. I feel good about this one.

**Pull request merged**